### PR TITLE
Task/96 - Adding the data module

### PR DESCRIPTION
**Pull request by forceindia712**
BuildConfig and Log not working

### CODE REVIEW

File: data/src/main/java/com/safetyheads/data/objects/YouTubeApi.kt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package com.safetyheads.akademiaandroida.network
+package com.safetyheads.data.objects
```
Comment (msarelo) [inline data/src/main/java/com/safetyheads/data/objects/YouTubeApi.kt:1]:
```suggestion
package com.safetyheads.data.network
```

File: data/src/main/java/com/safetyheads/data/repository/NetworkRepository.kt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package com.safetyheads.data.repository
```
Comment (msarelo) [inline data/src/main/java/com/safetyheads/data/repository/NetworkRepository.kt:1]:
```suggestion
package com.safetyheads.data.network.repository
```

File: data/src/main/java/com/safetyheads/data/retrofit/ApiClient.kt [hunk 3]
```diff
@@ -1,7 +1,6 @@
-package com.safetyheads.akademiaandroida.network
+package com.safetyheads.data.retrofit
```
Comment (msarelo) [inline data/src/main/java/com/safetyheads/data/retrofit/ApiClient.kt:1]:
```suggestion
package com.safetyheads.data.network.retrofit
```

File: data/src/main/java/com/safetyheads/data/services/YouTubeService.kt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package com.safetyheads.data.services
```
Comment (msarelo) [inline data/src/main/java/com/safetyheads/data/services/YouTubeService.kt:1]:
```suggestion
package com.safetyheads.data.network.services
```

File: data/src/test/java/com/safetyheads/domain/ExampleUnitTest.kt [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package com.safetyheads.domain
```
Comment (msarelo) [inline data/src/test/java/com/safetyheads/domain/ExampleUnitTest.kt:1]:
```suggestion
package com.safetyheads.data
```

**Review by msarelo — CHANGES_REQUESTED** [review]

File: data/src/androidTest/java/com/safetyheads/data/ExampleInstrumentedTest.kt
Comment (rafalkrystek) [inline data/src/androidTest/java/com/safetyheads/data/ExampleInstrumentedTest.kt:1]: Is this necessary? looks like android studio added test by default

File: data/src/test/java/com/safetyheads/data/ExampleUnitTest.kt
Comment (rafalkrystek) [inline data/src/test/java/com/safetyheads/data/ExampleUnitTest.kt:1]: Is this necessary? Looks like android studio added test by default.

**Review by rafalkrystek — COMMENTED** [review]
Check code formatting for example in here: 
                    Log.d(tag, message)
                }
// (empty line)
            }
            
            
class ApiClient(private val isDebug:Boolean, private val appLogger: AppLogger) {
// (missing space)

Use CTRL + ALT + L and CTRL + ALT + O

File: data/src/main/java/com/safetyheads/data/network/services/YouTubeService.kt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.safetyheads.data.network.services
+
+interface YouTubeService {
```
Comment (przemek19980102) [inline data/src/main/java/com/safetyheads/data/network/services/YouTubeService.kt:3]: is this expected to be empty?

File: data/src/main/java/com/safetyheads/data/network/retrofit/RetrofitInterceptor.kt [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package com.safetyheads.data.network.retrofit
+
+import okhttp3.Interceptor
+import okhttp3.Response
+
+class RetrofitInterceptor(private val appLogger: AppLogger) : Interceptor {
+    override fun intercept(chain: Interceptor.Chain): Response {
```
Comment (przemek19980102) [inline data/src/main/java/com/safetyheads/data/network/retrofit/RetrofitInterceptor.kt:7]: do we need it? as well as AppLogger? There is a library for that already https://stackoverflow.com/a/33256827

**Review by przemek19980102 — CHANGES_REQUESTED** [review]

File: data/src/androidTest/java/com/safetyheads/data/ExampleInstrumentedTest.kt
Comment (forceindia712) [reply data/src/androidTest/java/com/safetyheads/data/ExampleInstrumentedTest.kt:1]: When creating modules, classes for tests create themselves.

**Review by przemek19980102 — COMMENTED** [review]
Fixes to my comments are looking good

**Review by msarelo — APPROVED** [review]

**Pull request merged**